aten/src/ATen/CPUGeneratorImpl.cpp

```diff
@@ -131,69 +131,25 @@ uint64_t CPUGeneratorImpl::seed() {
 
 /**
  * Sets the internal state of CPUGeneratorImpl. The new internal state
- * must be a strided CPU byte tensor and of the same size as either
- * CPUGeneratorImplStateLegacy (for legacy CPU generator state) or
- * CPUGeneratorImplState (for new state).
- *
- * FIXME: Remove support of the legacy state in the future?
+ * must be a strided CPU byte tensor and of the same size as CPUGeneratorImplState.
  */
 void CPUGeneratorImpl::set_state(const c10::TensorImpl& new_state) {
   using detail::CPUGeneratorImplState;
   using detail::CPUGeneratorImplStateLegacy;
 
-  static_assert(std::is_standard_layout_v<CPUGeneratorImplStateLegacy>, "CPUGeneratorImplStateLegacy is not a PODType");
   static_assert(std::is_standard_layout_v<CPUGeneratorImplState>, "CPUGeneratorImplState is not a PODType");
-
-  static const size_t size_legacy = sizeof(CPUGeneratorImplStateLegacy);
-  static const size_t size_current = sizeof(CPUGeneratorImplState);
-  static_assert(size_legacy != size_current, "CPUGeneratorImplStateLegacy and CPUGeneratorImplState can't be of the same size");
+  static constexpr size_t size = sizeof(CPUGeneratorImplState);
```

Comment:
This isn't used as an lvalue so you can just do
```suggestion
  constexpr size_t size = sizeof(CPUGeneratorImplState);
```
for a lot simpler codegen

Reply:
Thank you, make a lot sense.